store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -246,8 +246,8 @@ public CarbonWriterBuilder withLoadOptions(Map<String, String> options) {
     Objects.requireNonNull(options, "Load options should not be null");
     //validate the options.
     if (options.size() > 9) {
-      throw new IllegalArgumentException("Supports only nine options now. "
-          + "Refer method header or documentation");
+      throw new IllegalArgumentException(
+          "Supports only ten options now. Refer method header or documentation");
```

Comment:
I think validating in this way is not good. @KanakaKumar can you have a check?

Reply:
Yes this number based validation needs to be removed. Only validation of given names already present in next for loop. If the original intention is to validate duplicate option entries, can have different logic and validation message.